app/screens/apps_form/apps_form_container.tsx

```diff
@@ -86,25 +82,21 @@ export default class AppsFormContainer extends PureComponent<Props, State> {
             };
         }
 
-        try {
-            const res = await this.props.actions.doAppCall({
-                ...call,
-                type: AppCallTypes.FORM,
-                values: {
-                    name: field.name,
-                    values,
-                    value,
-                },
-            });
-
-            if (res?.data?.form) {
-                this.setState({form: res.data.form});
-            }
+        const res = await this.props.actions.doAppCall({
+            ...call,
+            type: AppCallTypes.FORM,
+            values: {
+                name: field.name,
+                values,
+                value,
+            },
+        });
 
-            return res;
-        } catch (e) {
-            return makeError(e.message);
+        if (res?.data?.form) {
```

Comment:
Interesting. This is a TS method for the following? 
`if (res && res.data && res.data.form) {
`}

Reply:
@jfrerich That is the equivalent line of code. I think it's babel doing it for us, but 
typescript may also support this natively https://babeljs.io/docs/en/babel-plugin-proposal-optional-chaining